src/petro/extractor/looper_layer.hpp

```diff
@@ -32,16 +32,22 @@ class looper_layer : public Super
         m_loop = false;
     }
 
-    /// Returns the samples timestamp, offset based on the looping
-    uint64_t timestamp() const
+    /// Returns the sample's decoding timestamp with the loop offset
+    uint64_t decoding_timestamp() const
     {
-        return m_loop_offset + Super::timestamp();
+        return m_loop_offset + Super::decoding_timestamp();
+    }
+
+    /// Returns the sample's presentation timestamp with the loop offset
+    uint64_t presentation_timestamp() const
+    {
+        return m_loop_offset + Super::presentation_timestamp();
     }
 
     /// Advances to the next sample
     void advance()
     {
-        auto timestamp = Super::timestamp();
+        auto timestamp = Super::decoding_timestamp();
```

Comment:
This timestamp is used to increment the loop offset when we are at the end. This is the whole reason why i created that horrible ``timestamp()`` function.
I couldn't decide if I should use the ``presentation_timestamp`` or the ``decoding_timestamp`` for the offset incrementation. Do you have any thoughts on this? And if so could you maybe include a comment describing why the decoding_time should be used in favor of the presentation_time?

I thought about maybe having two offsets as well but that seemed a little weird.